### ISSUE TITLE
Appending to a sentence that begins with a quotation inserts an extra quotation mark

### ISSUE THREAD

**Issue by Gardamuse** [spencermountain/compromise#864]
I'm trying to insert some text at the end of each sentence, but sentences that begin with a quote has an extra quotation mark inserted at the beginning of the inserted text.

### Steps to Reproduce
```js
let doc = nlp('"Good bye," he said.')

doc.sentences().forEach((match) => {
   match.append('and left')
})
```

### Expected Result
`"Good bye," he said and left.`

### Actual Result
`"Good bye," he said "and left.`

### Additional Details
The same happens if the entire sentence is a quote.

**Input**: `"Hello there, he said."`
**Expected**: `"Hello there, he said and left."`
**Actual**: `"Hello there, he said "and left."`

This seems like a bug, though perhaps there is a better way to insert at the end of each sentence that what I've been trying.

**Comment by spencermountain** [comment]
Definitely a bug!
Sorry a out that. You're doing it right, and i will try to add a fix for this in v14.
Cheers

**Comment by spencermountain** [comment]
hey @Gardamuse this is fixed now in `14.4.1`:
```js
let doc = nlp('"Good bye," he said.')
doc.sentences().forEach((match) => {
  match.append('and left')
})
doc.text()// `"Good bye," he said and left.`
```
cheers
